dspace-api/src/test/java/org/dspace/content/EntityServiceImplTest.java

```diff
@@ -0,0 +1,224 @@
+        EntityTypeService entityTypeService = mock(EntityTypeService.class);
+        Item item = mock(Item.class);
+        List<MetadataValue> list = new ArrayList<>();
+        MetadataValue metadataValue = mock(MetadataValue.class);
+        list.add(metadataValue);
+        EntityType entityType = entityTypeService.findByEntityType(context, "testType");
+        when(metadataValue.getValue()).thenReturn("testType");
+        when(itemService.getMetadata(item, "relationship", "type", null, Item.ANY)).thenReturn(list);
+        assertEquals("TestGetType 0", entityType, entityService.getType(context, entity));
+    }
+
+    @Test
+    public void testGetLeftRelation() {
+        Item item = mock(Item.class);
+        UUID uuid = UUID.randomUUID();
+        Relationship relationship = mock(Relationship.class);
+        List<Relationship> relationshipList = new ArrayList<>();
+        relationshipList.add(relationship);
+        Entity entity = mock(Entity.class);
+        when(entity.getItem()).thenReturn(item);
+        when(relationship.getLeftItem()).thenReturn(item);
+        when(item.getID()).thenReturn(uuid);
+        when(entity.getRelationships()).thenReturn(relationshipList);
+        assertEquals("TestGetLeftRelations 0", relationshipList, entityService.getLeftRelations(context, entity));
+    }
+
+    @Test
+    public void testGetRightRelation() {
+        Item item = mock(Item.class);
+        UUID uuid = UUID.randomUUID();
+        Relationship relationship = mock(Relationship.class);
+        List<Relationship> relationshipList = new ArrayList<>();
+        relationshipList.add(relationship);
+        Entity entity = mock(Entity.class);
+        when(entity.getItem()).thenReturn(item);
+        when(relationship.getRightItem()).thenReturn(item);
+        when(item.getID()).thenReturn(uuid);
+        when(entity.getRelationships()).thenReturn(relationshipList);
+        assertEquals("TestGetLeftRelations 0", relationshipList, entityService.getRightRelations(context, entity));
+    }
+
+    @Test
+    public void testGetRelationsByLabel() throws Exception {
+        Relationship relationship = mock(Relationship.class);
+        RelationshipType relationshipType = mock(RelationshipType.class);
+        relationship.setRelationshipType(relationshipType);
+        List<Relationship> relationshipList = new ArrayList<>();
+        relationshipList.add(relationship);
+        when(relationshipService.findAll(context)).thenReturn(relationshipList);
+        when(relationship.getRelationshipType()).thenReturn(relationshipType);
+        when(relationshipType.getLeftLabel()).thenReturn("leftLabel");
+        when(relationshipType.getRightLabel()).thenReturn("rightLabel");
+        assertEquals("TestGetRelationsByLabel 0", relationshipList,
+                entityService.getRelationsByLabel(context, "leftLabel"));
+    }
+
+    @Test
+    public void testGetAllRelationshipTypes() throws Exception {
```

Comment:
@cwilper : a part of me hates to ask this, as I appreciate that you've written all these unit tests.  But, could you add in some more inline comments into the larger test methods.  They are very difficult to read / understand without empty lines between sections or human readable explanations as to what you are mocking (and why), etc.  For example, the number of `when()` statements here is quite large (11), and I honestly don't understand all of them.